src/targets/gpu/device/prefix_scan_sum.cpp

```diff
@@ -24,12 +51,32 @@ void prefix_scan_sum(hipStream_t stream, const argument& result, const argument&
                         k[axis] = j;
                         return k;
                     };
-                    block_scan<block_size>(idx,
-                                           sum{},
-                                           0,
-                                           n,
-                                           [&](auto j) { return input[compute_idx(j)]; },
-                                           [&](auto j, auto x) { output[compute_idx(j)] = x; });
+                    if(reverse)
+                    {
+                        block_scan<block_size>(
+                            idx,
+                            sum{},
+                            0,
+                            n,
+                            reverse_scan(n,
+                                         [&](auto j) {
+                                             return exclusive ? output[compute_idx(j)]
+                                                              : input[compute_idx(j)];
+                                         }),
+                            reverse_scan(n, [&](auto j, auto x) { output[compute_idx(j)] = x; }));
```

Comment:
Isn't exclusive an adjustment to the output not the input? I would think it would be done something like this:

```cpp
block_scan<block_size>(
    idx,
    sum{},
    0,
    n,
    reverse_scan(n,
                 [&](auto j) {
                     input[compute_idx(j)];
                 }),
    reverse_scan(n, [&](auto j, auto x) {
        if (exclusive) {
            auto idx = j + 1;
            if (j == 0)
                output[compute_idx(0)] = 0;
            if (idx < n)
                output[compute_idx(idx)] = x;
        } else {
            output[compute_idx(j)] = x; 
        }
    }));
```